stockexchange/bot/management/commands/bot.py

```diff
@@ -0,0 +1,129 @@
+from django.core.management.base import BaseCommand
+from django.conf import settings
+import telebot
+from polygon import RESTClient
+from ...models import User, Portfolio
+import datetime
+from datetime import date
+from telebot import types
+import locale
+
+bot = telebot.TeleBot(settings.TOKEN)
+
+
+class Command(BaseCommand):
+    help = 'Stock exchange bot'
+
+    def handle(self, *args, **options):
+        bot.infinity_polling()
+
+
+@bot.message_handler(commands=["start"])
+def start(m):
+    u = User.objects.filter(id=m.chat.id)
+    if not u:
+        u = User(id=m.chat.id)
+        u.firstname = m.chat.first_name
+        u.secondname = m.chat.last_name
+        u.username = m.chat.username
+        try:
+            u.save()
+        except:
+            print(f'An error occurred while saving the user to the database: id {u.id}, username {u.username}')
+    else:
+        u = u[0]
+    markup = types.ReplyKeyboardRemove(selective=False)
+    bot.send_message(m.chat.id,
+                     f"Hello {u.firstname if u.firstname is not None else u.username}! \nYou can type any ticker and I will give you info",
+                     reply_markup=markup)
+
+
+@bot.message_handler(content_types=['text'])
+def text_message(message: str):
+    text = message.text.upper()
+    resp = request_ticker(text)
+    if resp:
+        markup = telebot.types.InlineKeyboardMarkup(row_width=1)
+        markup.add(telebot.types.InlineKeyboardButton(text='1 day', callback_data=f'{text}|day'))
+        markup.add(telebot.types.InlineKeyboardButton(text='1 week', callback_data=f'{text}|week'))
+        markup.add(telebot.types.InlineKeyboardButton(text='1 month', callback_data=f'{text}|month'))
+        markup.add(telebot.types.InlineKeyboardButton(text='1 year', callback_data=f'{text}|year'))
+        desc = "".join(["*%s*, " % i for i in resp.similar])
+        bot.send_message(message.chat.id, text=f"*{resp.name}* | {resp.exchange} | {resp.sector} \n"
+                                               f"_{resp.description}_ \n\n"
+                                               f"Peers {desc} \n\n"
+                                               f"*For what period would you like to receive information:*\n",
+                         reply_markup=markup, parse_mode='Markdown')
+
+    else:
+        bot.send_message(message.chat.id, 'Sorry, but the ticker was not found')
```

Comment:
There is a practice called "early return". Use it to avoid "spaghetti code":
```python
def foo():
    if a:
        code1
        code1
        code1
        code1
        code1
    else:
        code2
```
can be replaces with:
```python
def foo():
    if not a:
        code2
        return

    code1
    code1
    code1
    code1
    code1
```

code1 is main logic code and now it is on 1st level of function